charts/datadog/templates/_container-trace-agent.yaml

```diff
@@ -67,7 +71,10 @@
     - name: runtimesocket
       mountPath: {{ template "datadog.dockerOrCriSocketPath" . }}
     {{- end }}
-    {{- if .Values.datadog.apm.useSocketVolume }}
+    - name: dsdsocket
+      mountPath: {{ (dir .Values.datadog.dogstatsd.socketPath) }}
+    {{- if and .Values.datadog.apm.useSocketVolume (ne (dir .Values.datadog.dogstatsd.socketPath) (dir .Values.datadog.apm.socketPath)) }}
+      readOnly: true
```

Comment:
Technically I think `readOnly: true` could be set even if `.Values.datadog.apm.useSocketVolume` is false (the only condition that matters here is `ne (dir .Values.datadog.dogstatsd.socketPath) (dir .Values.datadog.apm.socketPath)`).

Reply:
I think we are still missing a corner case, when `datadog.apm.useSocketVolume` not `true` but `datadog.dogstatsd.socketPath != datadog.apm.socketPath`

```yaml
    - name: dsdsocket
      mountPath: {{ (dir .Values.datadog.dogstatsd.socketPath) }}
    {{- if not .Values.datadog.apm.useSocketVolume) }}
      readOnly: true
    {{- else if and .Values.datadog.apm.useSocketVolume (ne (dir .Values.datadog.dogstatsd.socketPath) (dir .Values.datadog.apm.socketPath)) }}
      readOnly: true
    - name: apmsocket
      mountPath: {{ (dir .Values.datadog.apm.socketPath) }}
    {{- end }}
```